BitFaster.Caching.UnitTests/Lru/LruBuilderTests.cs

```diff
@@ -86,5 +122,25 @@ public void TestConcurrencyLevel()
 
             constructor.Should().Throw<ArgumentOutOfRangeException>();
         }
+
+        [Fact]
+        public void ScopedPOC()
+        {
+            // Choose from 16 combinations of Lru/TLru, Instrumented/NotInstrumented, Atomic create/not atomic create, scoped/not scoped
+
+            // layer 1: can choose ConcurrentLru/TLru, FastConcurrentLru/FastConcurrentTLru 
+            var c = new ConcurrentLru<int, AsyncAtomic<int, Scoped<Disposable>>>(3);
```

Comment:
This doesn't work for IDisposable.

Cache sees AsyncAtomic, which is not IDisposable - thus the inner scoped item will never be disposed.

There should probably be DisposableAtom. Somehow this must handle races between IsValueCreated and Dispose.